src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -391,9 +391,13 @@ private JObject FilterRuntimes(JObject projectRoot, HashSet<string> excludedRIDs
 
         private static void WriteProject(JObject projectRoot, string projectJsonPath)
         {
-            string projectJson = JsonConvert.SerializeObject(projectRoot, Formatting.Indented);
-            Directory.CreateDirectory(Path.GetDirectoryName(projectJsonPath));
-            File.WriteAllText(projectJsonPath, projectJson + Environment.NewLine); 
+            string projectJson = JsonConvert.SerializeObject(projectRoot, Formatting.Indented) + Environment.NewLine;
+
+            if (!File.Exists(projectJsonPath) || !projectJson.Equals(File.ReadAllText(projectJsonPath)))
```

Comment:
Determining the correct set of inputs might be a bit of a chore since the inputs depend on

- the original project.json
- an optional versions file
- an optional package drop with varying packages

I think it's simpler, and more reliable, to make this determination in code.

I wouldn't imagine project.json files are that large and likely to lead to memory issues.  I opted for this vs an alternative line by line comparison because it seemed it would be more performant (in my mind).  Obviously, if memory consumption is an issue then perf goes out the window.  I'll have to rely on your experience here if you believe that string comparing project.json files is likely to cause problems.